src/js/utils/element-utils.js

```diff
@@ -59,11 +63,69 @@ export function getOrInsertCSSRule(styleParent, selectorText) {
     return {
       style: {
         setProperty: () => {},
-        removeProperty: () => {}
-      }
+        removeProperty: () => {},
+      },
     };
   }
 
   style.sheet.insertRule(`${selectorText}{}`, style.sheet.cssRules.length);
   return style.sheet.cssRules[style.sheet.cssRules.length - 1];
 }
+
+/**
+ * Gets the number represented by the attribute
+ * @param {HTMLElement} el
+ * @param {string} attrName
+ * @returns {number | undefined} Will return undefined if no attribute set
+ */
+export function getNumericAttr(el, attrName) {
+  const attrVal = el.getAttribute(attrName);
+  return attrVal != null ? +attrVal : undefined;
+}
+
+/**
+ * @param {HTMLElement} el
+ * @param {string} attrName
+ * @param {number} value
+ */
+export function setNumericAttr(el, attrName, value) {
+  // avoid setting a value that hasn't changed
+  if (getNumericAttr(el, attrName) == value) return;
+
+  // also handles undefined
+  if (value == null) {
+    el.removeAttribute(attrName);
+    return;
+  }
+
+  // force to a numberic value before setting as a string
+  el.setAttribute(attrName, `${+value}`);
+}
+
+/**
+ * @param {HTMLElement} el
+ * @param {string} attrName
+ * @returns {boolean}
+ */
+export function getBooleanAttr(el, attrName) {
+  return el.hasAttribute(attrName);
+}
+
+/**
+ *
+ * @param {HTMLElement} el
+ * @param {string} attrName
+ * @param {boolean} show
+ */
+export function setBooleanAttr(el, attrName, show) {
+  // avoid setting a value that hasn't changed
+  if (getBooleanAttr(el, attrName) == show) return;
+
+  // also handles undefined
+  if (show == null) {
+    el.removeAttribute(attrName);
+    return;
+  }
+
+  el.toggleAttribute(attrName, show);
```

Comment:
I don't think this is right? And the logic is kind of confusing. The second arg of toggleAttribute is `force`, so I'm not clear how this is intended to work.

I'd recommend changing "show" to "value", and then if value is falsey removeAttribute, if truthy `setAttribute(attrName, attrName).

Reply:
should this call it `force` to match the name of `toggleAttribute`? I think `show` would be a fine name for the setters that use this helper, though.

Might be worth also adding a note in the jsdoc about what this means (or even linking to mdn or something)